src/proxy/topic/TopicProxy.ts

```diff
@@ -0,0 +1,123 @@
+import {ITopic} from '../ITopic';
+import {Message, MessageListener} from '../MessageListener';
+import {AddressImpl, HazelcastError, UUID} from '../../core';
+import {TopicAddMessageListenerCodec} from '../../codec/TopicAddMessageListenerCodec';
+import {TopicRemoveMessageListenerCodec} from '../../codec/TopicRemoveMessageListenerCodec';
+import { ListenerMessageCodec} from '../../listener/ListenerMessageCodec';
+import {ClientMessage} from '../../protocol/ClientMessage';
+import {assertNotNull} from '../../util/Util';
+import {PartitionSpecificProxy} from '../PartitionSpecificProxy';
+import {TopicPublishCodec} from '../../codec/TopicPublishCodec';
+import Long = require('long');
+import {Data} from '../../serialization';
+import {ClientConfig, ClientConfigImpl} from '../../config';
+import {ProxyManager} from '../ProxyManager';
+import {PartitionService} from '../../PartitionService';
+import {InvocationService} from '../../invocation/InvocationService';
+import {SerializationService} from '../../serialization/SerializationService';
+import {ListenerService} from '../../listener/ListenerService';
+import {ClusterService} from '../../invocation/ClusterService';
+import {ConnectionRegistry} from '../../network/ConnectionRegistry';
+import {SchemaService} from '../../serialization/compact/SchemaService';
+import {Connection} from '../../network/Connection';
+import {TopicOverloadPolicy} from '../TopicOverloadPolicy';
+
+export class TopicProxy<E> extends PartitionSpecificProxy implements ITopic<E> {
+
+
+    private readonly overloadPolicy: TopicOverloadPolicy;
+    private readonly localAddress: AddressImpl;
```

Comment:
Why do we have them if they are not used?

Reply:
I would fix them after major things are done. Got them.